### PR TITLE
FOUR-8725 design and move the autovalidate control

### PR DESCRIPTION
**Pull request by devmiguelangel**
## Issue & Reproduction Steps

Expected behavior: 
Improve the auto-validate control design and move it to the top rail

Actual behavior: 
N/A

## Solution
Re-design and implement the functionalities to the new auto-validate control

## How to Test
Test the steps above

1. Open a BPMN process
2. It should display the new auto-validate control in the top rail of the screen
3. Click on the new auto-validate control in the top rail
4. It should display the issue button next to the auto-validate control
5. Click on this button to show the detailed panel

## Related Tickets & Packages
[FOUR-8725](https://processmaker.atlassian.net/browse/FOUR-8725)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

[FOUR-8725]: https://processmaker.atlassian.net/browse/FOUR-8725?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/topRail/TopRail.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import store from '@/store';
+import { ValidateButton, ValidateIssue, ValidatePanel } from '@/components/topRail/validateControl';
+
+export default {
+  components: {
+    ValidateButton,
+    ValidateIssue,
+    ValidatePanel,
+  },
+  props: ['validationErrors', 'warnings'],
```
Comment (rodriquelca) [inline src/components/topRail/TopRail.vue:29]: please define type of the props.
example 
```
props: {
    title: String,
    likes: Number
  }
```

File: src/components/topRail/TopRail.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  watch: {
+    numberOfErrors(newValue) {
+      // Checks the number of errors, if it is "0" hides the panel errors
+      if (newValue === 0) {
+        this.isOpenPanel = false;
+      }
+    },
+  },
+  methods: {
+    handleOpenIssue(value) {
```
Comment (rodriquelca) [inline src/components/topRail/TopRail.vue:62]: add documentations for the methods use JDoc

File: src/components/topRail/TopRail.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      // Show/hide the issue button
+      this.isOpenIssue = value;
+      // Set the auto-validate value store
+      store.commit('setAutoValidate', this.isOpenIssue);
+    },
+    handleOpenPanel(value) {
```
Comment (rodriquelca) [inline src/components/topRail/TopRail.vue:68]: add documentations for the methods use JDoc

File: src/components/topRail/TopRail.vue (hunk 1 above)
Comment (devmiguelangel) [reply src/components/topRail/TopRail.vue:29]: It will be fixed

File: src/components/topRail/TopRail.vue (hunk 3 above)
Comment (devmiguelangel) [reply src/components/topRail/TopRail.vue:68]: It will be added

File: src/components/toolbar/ToolBar.vue [hunk 4]
```diff
@@ -4,9 +4,13 @@
       aria-label="Toolbar" :class="{ 'ignore-pointer': canvasDragPosition }"
     >
       <breadcrumb :breadcrumb-data="breadcrumbData" />
-      <div class="mr-3">
-        <align-buttons @save-state="$emit('save-state')" />
+      <div class="mr-3" style="display: flex;">
```
Comment (josechirivella) [inline src/components/toolbar/ToolBar.vue:7]: There should be a class coming from Bootstrap to apply display flex here. Avoid inline styling

File: src/components/topRail/topRail.scss [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.top-rail-container {
+  position: relative;
+  display: flex;
+  align-items: center;
+  padding: {
+    left: 8px;
+    right: 8px;
+  };
```
Comment (josechirivella) [inline src/components/topRail/topRail.scss:8]: this is incorrect. should be `padding: 0 8xp 0 8px;` or `padding: 0 8px`

**Review by josechirivella — CHANGES_REQUESTED** [review]

File: src/components/topRail/topRail.scss (hunk 5 above)
Comment (devmiguelangel) [reply src/components/topRail/topRail.scss:8]: It will be fixed

**Review by danloa — APPROVED** [review]

**Pull request merged**